libs/pika/init_runtime/include/pika/init_runtime/init_runtime.hpp

```diff
@@ -133,19 +84,110 @@ namespace pika {
         init_params const& params = init_params());
     PIKA_EXPORT int init(std::nullptr_t, int argc, const char* const* argv,
         init_params const& params = init_params());
-    PIKA_EXPORT bool start(std::function<int(pika::program_options::variables_map&)> f, int argc,
+
+    /// Start the runtime.
+    ///
+    /// @param f entry point of the first task on the pika runtime. f will be passed all non-pika
+    /// command line arguments.
+    /// @param argc number of arguments in argv
+    /// @param argv array of arguments. The first element is ignored.
+    ///
+    /// @pre `(argc == 0 && argv == nullptr) || (argc >= 1 && argv != nullptr)`
+    /// @pre the runtime is stopped
+    /// @post the runtime is running
+    PIKA_EXPORT void start(std::function<int(pika::program_options::variables_map&)> f, int argc,
         const char* const* argv, init_params const& params = init_params());
-    PIKA_EXPORT bool start(std::function<int(int, char**)> f, int argc, const char* const* argv,
+
+    /// Start the runtime.
+    ///
+    /// @param f entry point of the first task on the pika runtime. f will be passed all non-pika
+    /// command line arguments.
+    /// @param argc number of arguments in argv
+    /// @param argv array of arguments. The first element is ignored.
+    ///
+    /// @pre `(argc == 0 && argv == nullptr) || (argc >= 1 && argv != nullptr)`
+    /// @pre the runtime is stopped
+    /// @post the runtime is running
+    PIKA_EXPORT void start(std::function<int(int, char**)> f, int argc, const char* const* argv,
         init_params const& params = init_params());
-    PIKA_EXPORT bool start(std::function<int()> f, int argc, const char* const* argv,
+
+    /// Start the runtime.
+    ///
+    /// @param f entry point of the first task on the pika runtime
+    /// @param argc number of arguments in argv
+    /// @param argv array of arguments. The first element is ignored.
+    ///
+    /// @pre `(argc == 0 && argv == nullptr) || (argc >= 1 && argv != nullptr)`
+    /// @pre the runtime is not running
+    PIKA_EXPORT void start(std::function<int()> f, int argc, const char* const* argv,
         init_params const& params = init_params());
-    PIKA_EXPORT bool start(std::nullptr_t, int argc, const char* const* argv,
+
+    [[deprecated]] PIKA_EXPORT void start(std::nullptr_t, int argc, const char* const* argv,
```

Comment:
To do: deprecate or remove? Remove docstrings from here as well (they are not included in the generated documentation)?

Reply:
Keeping, but leaving undocumented.